params/json.libevm.go

```diff
@@ -124,3 +124,27 @@ func toJSONRawMessages(v any) (map[string]json.RawMessage, error) {
 	}
 	return msgs, nil
 }
+
+// UnmarshalChainConfig JSON decodes the given data into the config and the extra
+// pointer arguments. The root JSON keys except the "extra" key are decoded into
+// the config argument, and the object at the "extra" key, if present, is decoded
+// into the extra argument. Note the extra argument must be a non-nil pointer or
+// an error would be returned.
+func UnmarshalChainConfig(data []byte, config *ChainConfig, extra any) (err error) {
```

Comment:
We should probably call it `UnmarshalChainConfigJSON` to be precise. Sorry, I know I gave that example over DM.